### PR TITLE
[#6455]: Updated extension management

### LINKED ISSUE #6455: Allow multiple extensions installation
### User Story

As a user, I want to be able to install multiple extensions at the same time.

### Description

*Type*: Feature

Only 1 extension can be built at a time. Relax this constraint.

This includes:

* using the extension URL as unique ID
* making sure that new extensions using already installed hook ids will be rejected (*warning* one extension can define multiple hooks)
* allowing users to uninstall existing extension (with a swipe, like chat in home screen)
* fix `store/` events so that they are properly per extension isolated

### PR DESCRIPTION
**Pull request by alwx**
Fixes #6455.
Also here: https://github.com/status-im/pluto/pull/88

This PR improves the extension management mechanism — allows to remove extensions and also moves registry from `pluto` to `status-react`.

### CODE REVIEW

File: src/status_im/extensions/registry.cljs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                       :follow-up  :extensions/stage}}))
+
+(fx/defn initialize
+  [{{:account/keys [account]} :db}]
+  (let [{:keys [extensions dev-mode?]} account]
+    (when dev-mode?
+      {:extensions/load {:extensions (vals extensions)
+                         :follow-up  :extensions/add-to-registry}})))
+
+(defn existing-commands
```
Comment (jeluard) [inline src/status_im/extensions/registry.cljs]: I'd rather use hooks here, not commands

Comment (alwx) [reply src/status_im/extensions/registry.cljs]: you're right, will change it

Comment (jeluard) [reply src/status_im/extensions/registry.cljs]: Actually the implementation should be a bit more generic and consider all hook types, not just :commands

File: src/status_im/extensions/registry.cljs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+(defn existing-hooks
+  [cofx extension-data]
+  (let [added-hooks (reduce (fn [acc [_ {:keys [hooks] :as v}]]
+                              (->> (:commands hooks)
+                                   (keys)
+                                   (into acc)))
+                            #{}
+                            (get-in cofx [:db :account/account :extensions]))
+        hooks       (->> (get-in extension-data [:data :hooks :commands])
```
Comment (jeluard) [inline src/status_im/extensions/registry.cljs]: `:commands` should not be hardcoded, all hook types must be considered.
But the id is unique per hook type, not globally.

**Comment by alwx** [comment]
@jeluard updated it already!

**Review by jeluard — APPROVED** [review]

File: src/status_im/extensions/registry.cljs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            [status-im.accounts.update.core :as accounts.update]
+            [status-im.i18n :as i18n]
+            [status-im.utils.fx :as fx]
+            [clojure.set :as set]))
+
+(fx/defn update-hooks
+  [{:keys [db] :as cofx} extension-key]
+  (let [{:keys [dev-mode?] :as account} (get db :account/account)
+        {:keys [active? hooks]} (get-in account [:extensions extension-key])]
+    (apply utils/merge-fx cofx
```
Comment (flexsurfer) [inline src/status_im/extensions/registry.cljs:16]: looks complex, it's difficult to read and understand what happens here

File: src/status_im/extensions/registry.cljs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    (fx/merge cofx
+              {:db (update-in db [:account/account :extensions extension-key] merge data)}
+              (update-hooks extension-key))))
+
+(fx/defn remove-from-registry
+  [{:keys [db] :as cofx} extension-key]
+  (fx/merge cofx
+            {:db (assoc-in db [:account/account :extensions extension-key :active?] false)}
+            (update-hooks extension-key)
+            {:db (update-in db [:account/account :extensions] dissoc extension-key)}))
```
Comment (flexsurfer) [inline src/status_im/extensions/registry.cljs:36]: this looks like black magic

**Review by flexsurfer — APPROVED** [review]

**Comment by jeluard** [comment]
@alwx Please resolve conflicts :sweat_smile:

File: src/status_im/extensions/registry.cljs (hunk 4 above)
Comment (alwx) [reply src/status_im/extensions/registry.cljs]: not anymore!

Comment (alwx) [reply src/status_im/extensions/registry.cljs]: not anymore! :)

Comment (alwx) [reply src/status_im/extensions/registry.cljs:36]: not anymore! :)

File: src/status_im/extensions/registry.cljs (hunk 3 above)
Comment (alwx) [reply src/status_im/extensions/registry.cljs:16]: simplified it.
however, this function was simply moved from `pluto` here and originally contained almost no changes.

**Pull request merged**